### PR TITLE
Image upload

### PR DESCRIPTION
**Pull request by Capocaccia**
Still working on tests and figuring out how to use the pull requests.  Its a bit of a mess.  Any advice on testing file upload functionality?  We generally know the things we need to test like file size and type but are confused about how to do it. For example, we are stuck on how to test the response this image submission without actually submitting an image.  Let us know if you would rather discuss this with us in person than over pull requests.   @sscoth

### CODE REVIEW

**Comment by sscotth** [comment]
There is no reference to the photo in the post show page. Make sure the show page displays the image if there is one. At a minimum that should be done.

As for your test, it doesn't work. We can talk about the test tomorrow.

Also, I'd like to even see a thumbnail get generated, but that can be a separate pull request. Look at this library for that: https://github.com/aheckmann/gm

**Comment by katyjustiss** [comment]
Ok, I added images onto the post show page and added the image upload capabilities onto the form on that page as well. I'll look into the thumbnail thing. And would be happy to meet to discuss testing.

File: app/post/show.jade [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+    label(for='text') Message:
     input(name='text')
+    label(for='image') Upload Image:
+    input(type='file' name='image' accept="image/*")
     input(type='submit')
 
 
   .post= post.text
+  if post.url
+    img.post(src=post.url)
```
Comment (sscotth) [inline app/post/show.jade]: You probably want to nest the image in the `<div class="post">` and you wont need the post class in the `` tag
Try:

``` jade
.post
  p= post.text
  if post.url
    img(src=post.url)
```

File: app/post/show.jade [hunk 2]
```diff
@@ -1,9 +1,14 @@
 extends ../templates/layout
 
 block content
-  form(action='/post', method='POST')
+  form(action='/post', method='POST', enctype="multipart/form-data")
```
Comment (sscotth) [inline app/post/show.jade]: stick with single quotes

File: app/post/show.jade [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
+    label(for='text') Message:
     input(name='text')
+    label(for='image') Upload Image:
+    input(type='file' name='image' accept="image/*")
```
Comment (sscotth) [inline app/post/show.jade]: single quotes & separate attributes with a comma

File: app/post/index.jade [hunk 4]
```diff
@@ -1,8 +1,11 @@
 extends ../templates/layout
 
 block content
-  form(action='/post', method='POST')
+  form(action='/post', method='POST', enctype="multipart/form-data")
+    label(for='text') Message:
     input(name='text')
+    label(for='image') Upload Image:
+    input(type='file' name='image' accept="image/*")
```
Comment (sscotth) [inline app/post/index.jade]: single quotes & separate attributes with a comma

File: app/post/index.jade [hunk 5]
```diff
@@ -1,8 +1,11 @@
 extends ../templates/layout
 
 block content
-  form(action='/post', method='POST')
+  form(action='/post', method='POST', enctype="multipart/form-data")
```
Comment (sscotth) [inline app/post/index.jade]: single quotes

File: app/imageUpload/imageController.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var fs = require('fs');
+var imgur = require('imgur');
+var multer = require('multer');
+
+var upload = multer({
+  dest: 'uploads/',
+  limits: {
+    fileSize: 10000000 //20MB
```
Comment (sscotth) [inline app/imageUpload/imageController.js]: is this 20MB?

maybe you should create a variable (no magic numbers)
`var tenMegaBytes = 1000000`

File: app/imageUpload/imageController.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  fileFilter: function (req, file, cb) {
+    cb(null, file.mimetype.slice(0,6) === 'image/');
+  }
+});
+
+var uploadMulter = upload.single('image')
+
+module.exports.img = function (req, res, next) {
```
Comment (sscotth) [inline app/imageUpload/imageController.js]: Can you come up with a better function name than "img"? Even `module.exports = function` would be preferred. Just change the routes.js file to match.

**Pull request closed without merging**